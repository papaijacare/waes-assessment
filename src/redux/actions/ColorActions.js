import { COLOR } from '../types'

export const setSelectedColor = color => ({
    type: COLOR.SET_SELECTED,
    payload: color
})

export const setFilteredColor = color => ({
    type: COLOR.SET_FILTERED,
    payload: color
})

export const removeFilteredColor = color => ({
    type: COLOR.REMOVE_FILTERED,
    payload: color
})