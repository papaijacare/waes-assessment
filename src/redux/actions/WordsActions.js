import { WORDS } from '../types'

export const addWord = (color, word) => ({
    type: WORDS.ADD_WORD,
    payload: { color, word }
})

export const removeWord = (color, word) => ({
    type: WORDS.REMOVE_WORD,
    payload: { color, word }
})