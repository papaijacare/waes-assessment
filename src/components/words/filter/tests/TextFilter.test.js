import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import configureStore from 'redux-mock-store'

import { TextFilter } from '../..'
import { setFilteredColor } from "../../../../redux/actions/ColorActions"

const mockStore = configureStore()
const store = mockStore({
    words: {
        red: ['WAES']
    },
    colors: { 
        filteredColors: [], 
        selectedColor: undefined
    }
})

const instance = () => {
    const props = {
        store: store,
    }
    const wrapper = shallow(<TextFilter { ...props }/>)
    
    return {
        wrapper
    }
}

describe('TextFilter', () => {
    test('renders correctly intiallly', () => {
        const { wrapper } = instance()
        const component = wrapper.dive()

        expect(toJson(component)).toMatchSnapshot()
    })

    test('renders correct when words are filtered', () => {
        store.dispatch(setFilteredColor('red'))

        const { wrapper } = instance()
        const component = wrapper.dive()

        expect(store.getActions()).toMatchSnapshot()
    })
})