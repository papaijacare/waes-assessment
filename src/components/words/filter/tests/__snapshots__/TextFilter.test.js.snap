// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFilter renders correct when words are filtered 1`] = `
Array [
  Object {
    "payload": "red",
    "type": "SET_FILTERED_COLOR",
  },
]
`

exports[`TextFilter renders correctly intiallly 1`] = `
<div>
  <LabelGroup>
     
  </LabelGroup>
</div>
`
